### PR TITLE
Add edit & delete functions to event card

### PR DESCRIPTION
**Pull request by djnunez-aot**
*Issue #:*
https://github.com/bcgov/met-public/issues/<Put the github issue number here>

-Work in progress

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the met-public license (Apache 2.0).

### CODE REVIEW

File: met-web/src/components/engagement/form/EngagementWidgets/Events/VirtualSessionFormDrawer.tsx [hunk 1]
```diff
@@ -31,29 +36,65 @@ type VirtualSessionForm = yup.TypeOf<typeof schema>;
...
+    }, [eventToEdit]);
+
     const { handleSubmit, reset } = methods;
 
-    const onSubmit: SubmitHandler<VirtualSessionForm> = async (data: VirtualSessionForm) => {
-        if (!widget) {
-            return;
+    const updateEvent = async (data: VirtualSessionForm) => {
+        if (eventToEdit && widget) {
+            console.log(eventToEdit);
```
Comment (jadmsaadaot) [inline met-web/src/components/engagement/form/EngagementWidgets/Events/VirtualSessionFormDrawer.tsx:78]: console log

File: met-api/src/met_api/models/event_item.py [hunk 2]
```diff
@@ -29,3 +31,36 @@ class EventItem(BaseModel):  # pylint: disable=too-few-public-methods, too-many-
     def save_event_items(cls, event_items: list) -> None:
         """Update widgets.."""
         db.session.bulk_save_objects(event_items)
+        
+    
+    @classmethod
+    def update_event_item(cls, event_data: dict) -> Optional[EventItem or DefaultMethodResult]:
+        """Update Event Item."""
+        query = EventItem.query.filter_by(id=event_data.get('id'))
+        update_fields = dict(
```
Comment (jadmsaadaot) [inline met-api/src/met_api/models/event_item.py:40]: If we're patching we don't want to ideally touch all fields when we update, we can pass only the affected fields. We can follow the pattern in the engagement model. edit_engagment is what we did so we don't do it the update_engagement way and send all fields to the update
image

File: met-api/src/met_api/models/event_item.py [hunk 3]
```diff
@@ -29,3 +31,36 @@ class EventItem(BaseModel):  # pylint: disable=too-few-public-methods, too-many-
...
+            start_date=event_data.get('start_date', None),
+            end_date=event_data.get('end_date', None),
+            url=event_data.get('url', None),
+            url_label=event_data.get('url_label', None),
+            sort_index=event_data.get('sort_index', None),
+            widget_events_id=event_data.get('widget_events_id', None),
+            updated_date=datetime.utcnow(),
+        )
+        if not query:
+            return DefaultMethodResult(False, 'Event Not Found', event_data.get('widgets_event_id'))
```
Comment (jadmsaadaot) [inline met-api/src/met_api/models/event_item.py:53]: I think we have moved away from DefaultMethodResult. If it's still used in the codebase flag it in the group so we create a task to remove them

File: met-api/src/met_api/models/event_item.py (hunk 2 above)
Comment (djnunez-aot) [reply met-api/src/met_api/models/event_item.py:40]: good idea there

File: met-api/src/met_api/models/event_item.py (hunk 3 above)
Comment (djnunez-aot) [reply met-api/src/met_api/models/event_item.py:53]: where would you like me to flag it

Comment (jadmsaadaot) [reply met-api/src/met_api/models/event_item.py:53]: If it's a lot, then we can just create a task to find and remove them, maybe there's one already?

**Review by jadmsaadaot — APPROVED** [review]
Good job!

**Pull request merged**